src/Authentication/Actions/Email2FA.php

```diff
@@ -71,14 +71,13 @@ public function handle(IncomingRequest $request)
 
         // Send the user an email with the code
         helper('email');
-        $return = emailer()->setFrom(setting('Email.fromEmail'), setting('Email.fromName') ?? '')
-            ->setTo($user->email)
-            ->setSubject(lang('Auth.email2FASubject'))
-            ->setMessage(view(setting('Auth.views')['action_email_2fa_email'], ['code' => $identity->secret]))
-            ->send();
-
-        if ($return === false) {
-            throw new RuntimeException('Cannot send email for user: ' . $user->email);
+        $email = emailer()->setFrom(setting('Email.fromEmail'), setting('Email.fromName') ?? '');
+        $email->setTo($user->email);
+        $email->setSubject(lang('Auth.email2FASubject'));
+        $email->setMessage(view(setting('Auth.views')['action_email_2fa_email'], ['code' => $identity->secret]));
+
+        if ($email->send(false) === false) {
+            throw new RuntimeException('Cannot send email for user: ' . $user->email . ' ' . $email->printDebugger(['headers']));
         }
```

Comment:
Is it better to call `$email->clear()` here?

Reply:
I don't think so, since it is not a loop, and the `$email` variables are not being reinitialized. See [docs](https://codeigniter4.github.io/CodeIgniter4/libraries/email.html#clear).

Reply:
The Email service is shared.
https://github.com/codeigniter4/shield/blob/44e65eff3043530e21337ffdb2d2f6c991a0e548/src/Helpers/email_helper.php#L44
If devs use the shared Email (`service('email')`) in their app, the state is not clear.

Reply:
Does 'clearing' not mean the same as trying to reinitialise the email variables?

Reply:
If necessary, probably a call to `clear()` should happen in the `emailer()` helper function.

Reply:
Done, thank you.

Reply:
No, it is not.
The issue is that a dev might use the Email object in their app after Shield send email.

`$email->send()` calls `clear()`. But you changed it to `$email->send(false)` that does not call `clear()`.
`emailer()` is a helper for Shield. It is not a framework helper, so a dev does not use it.

If a dev calls `service('email')`, the `clear()` you added in the `emailer()` is not called.
To begin with, `emailer()` calls `$email->initialize()` so you don't need to add `clear()`.

Reply:
`service('email')` or `Services::email()` returns a shared instance by default.
It might be not good design.

Getting a new instance in `emailer()` may be better.
```diff
--- a/src/Helpers/email_helper.php
+++ b/src/Helpers/email_helper.php
@@ -41,7 +41,7 @@ function emailer(array $overrides = []): Email
         }
 
         /** @var Email $email */
-        $email = service('email');
+        $email = service('email', false);
 
         return $email->initialize($config);
     }
```

Reply:
Oh, thank you for the explanation.

Reply:
Using a single service makes testing harder, just need to be sure to inject the right class ahead of time. I think calling `clear()` before the final return is fine, but I also expect developers to handle this themselves. Any time `email` service is used there is a possibility that it is not in a "clean" state so should be initialized/cleaned. This is of course terrible design but will be fixed when `Mailer` is released.

Reply:
Calling `clear()` before the final return does not make sense, becaue `initialize()` calls `clear()`.

Then, calling `clear()` after `throw new RuntimeException()` is the only way.
Otherwise this PR is a breaking change.